apps/docs/src/guide/provider/snippets/functionality/get-messages-by-nonce.ts

```diff
@@ -1,9 +1,27 @@
 // #region getMessageByNonce
-import { Provider } from 'fuels';
+import { launchTestNode } from 'fuels/test-utils';
 
-import { LOCAL_NETWORK_URL } from '../../../../env';
-
-const provider = await Provider.create(LOCAL_NETWORK_URL);
+const { provider } = await launchTestNode({
+  nodeOptions: {
+    snapshotConfig: {
+      stateConfig: {
+        messages: [
+          {
+            sender:
+              '0x22cae5308938e8b4caf217b6464884f6331eff05e81468df8ccd08126effc8d0',
+            recipient:
+              '0x8d2af98a4198732a46bf65d87a73427dd7608acaad2414585d8ccdd6f59c437b',
+            nonce:
+              '0x381de90750098776c71544527fd253412908dec3d07ce9a7367bd1ba975908a0',
+            amount: 100_000,
+            data: '',
+            da_height: 0,
+          },
+        ],
+      },
+    },
+  },
+});
```

Comment:
```suggestion
const message = new TestMessage({
  nonce: "0x381de90750098776c71544527fd253412908dec3d07ce9a7367bd1ba975908a0"
});

const { provider } = await launchTestNode({
  walletsConfig: {
    messages: [message],
  },
});
```

Would something like this be cleaner?

Reply:
https://github.com/FuelLabs/fuels-ts/pull/3420/commits/7dc91d71c8f1522d5ae5876ce0633cb29d935613